### PR TITLE
feat: add set approve for all screen

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR aims to reuse the token approve confirmation screen for `setapproveforall`.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2408

## **Manual testing steps**

1. Go to test dapp
2. Deploy ERC721
3. Mint 
4. Click `SET APPROVAL FOR ALL`
5. Confirm and check in the activity view if the transaction was a `set approval for all` and it's confirmed.

## **Screenshots/Recordings**

[set-approve-for-all.webm](https://github.com/MetaMask/metamask-mobile/assets/45455812/e64f5f27-b9c4-4c32-8573-52475a6f2ce4)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: e2e/specs/confirmations/set-approve-for-all-erc721.spec.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import { ActivitiesViewSelectorsText } from '../../selectors/ActivitiesView.selectors';
+import Assertions from '../../utils/Assertions';
+import { ContractApprovalModalSelectorsText } from '../../selectors/Modals/ContractApprovalModal.selectors';
+import ContractApprovalModal from '../../pages/modals/ContractApprovalModal';
+
+describe(SmokeConfirmations('ERC721 token'), () => {
+  const NFT_CONTRACT = SMART_CONTRACTS.NFTS;
+
+  beforeAll(async () => {
+    jest.setTimeout(150000);
```
Comment (matthewwalsh0) [inline e2e/specs/confirmations/set-approve-for-all-erc721.spec.js:24]: Do we still need this? Is this not at least centralised somewhere?

File: locales/languages/en.json [hunk 2]
```diff
@@ -1380,10 +1380,12 @@
     "approve": "Approve",
     "allow_to_access": "Give permission to access your",
     "allow_to_address_access": "Give this address access your",
+    "allow_to_transfer_all": "Allow access to and transfer of all your",
     "spend_cap": "Spending cap request for your",
     "token": "token",
     "nft": "NFT",
     "you_trust_this_site": "By granting permission, you're allowing the following third party to access your funds.",
+    "you_trust_this_third_party" : "This allows a third party to access and transfer the your NFTs without further notice until you revoke its access.",
```
Comment (matthewwalsh0) [inline locales/languages/en.json:1388]:
```suggestion
    "you_trust_this_third_party" : "This allows a third party to access and transfer your NFTs without further notice until you revoke its access.",
```

**Review by jpuri — COMMENTED** [review]
Code changes look good to me, some unit test coverage also in the PR will be great.

File: app/util/transactions/index.js [hunk 3]
```diff
@@ -1541,3 +1560,13 @@ export const getIsSwapTransaction = (data, origin, to, chainId) => {
...
 export const getIsNativeTokenTransferred = (txParams) =>
   txParams?.value !== '0x0';
+
+/**
+ * Checks if the given token standard is non-fungible (ERC721 or ERC1155).
+ *
+ * @param {string} tokenStandard - The token standard to check.
+ * @returns {boolean} - True if the token standard is ERC721 or ERC1155, otherwise false.
+ */
+export function isNonFungibleTokenStandard(tokenStandard) {
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply app/util/transactions/index.js]: Changed to `isNFTTokenStandard`.

File: app/util/transactions/index.js [hunk 4]
```diff
@@ -64,16 +70,19 @@ export const UNKNOWN_FUNCTION_KEY = 'unknownFunction';
...
 export const TRANSFER_FROM_FUNCTION_SIGNATURE = '0x23b872dd';
 export const APPROVE_FUNCTION_SIGNATURE = '0x095ea7b3';
 export const CONTRACT_CREATION_SIGNATURE = '0x60a060405260046060527f48302e31';
 export const INCREASE_ALLOWANCE_SIGNATURE = '0x39509351';
+export const SET_APPROVAL_FOR_ALL_SIGNATURE = '0xa22cb465';
 
 export const TRANSACTION_TYPES = {
   APPROVE: 'transaction_approve',
   INCREASE_ALLOWANCE: 'increaseAllowance',
+  SET_APPROVAL_FOR_ALL: 'setapprovalforall',
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply app/util/transactions/index.js]: Done, updated both to follow the standard.

File: e2e/specs/confirmations/set-approve-for-all-erc721.spec.js (hunk 1 above)
Comment (vinistevam) [reply e2e/specs/confirmations/set-approve-for-all-erc721.spec.js:24]: You are right, the global is in `.detoxrx.js` and it's greater than what I set.
I tested locally a couple of times.

**Review by jpuri — APPROVED** [review]
👍

**Comment by sleepytanya** [comment]
Works  beautifully! 
Samsung S24+, Android 14, Bitrise QA build https://app.bitrise.io/build/91102a1f-b17d-43cf-9c58-f2c66591c6b5


https://github.com/MetaMask/metamask-mobile/assets/104780023/1911d54d-0aaa-4342-82c0-23b200358237

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by vinistevam** [comment]
E2E error:

Error has been happening since [the commit ](https://github.com/MetaMask/metamask-mobile/commit/d5cecafb847199f0b4f58e242adf25fc55d6d300).

**Pull request merged**